pkg/trimaran/lowriskovercommitment/lowriskovercommitment.go

```diff
@@ -0,0 +1,290 @@
+	score = int64(math.Round(totalScore))
+	return score, framework.NewStatus(framework.Success, "")
+}
+
+// Name : name of plugin
+func (pl *LowRiskOverCommitment) Name() string {
+	return Name
+}
+
+// ScoreExtensions : an interface for Score extended functionality
+func (pl *LowRiskOverCommitment) ScoreExtensions() framework.ScoreExtensions {
+	return pl
+}
+
+// NormalizeScore : normalize scores
+func (pl *LowRiskOverCommitment) NormalizeScore(context.Context, *framework.CycleState, *v1.Pod, framework.NodeScoreList) *framework.Status {
+	return nil
+}
+
+// computeRank : rank function for the LowRiskOverCommitment
+func (pl *LowRiskOverCommitment) computeRank(metrics []watcher.Metric, nodeInfo *framework.NodeInfo, pod *v1.Pod,
+	podRequests *framework.Resource, podLimits *framework.Resource) float64 {
+	node := nodeInfo.Node()
+	// calculate risk based on requests and limits
+	nodeRequestsAndLimits := trimaran.GetNodeRequestsAndLimits(nodeInfo.Pods, node, pod, podRequests, podLimits)
+	riskCPU := pl.computeRisk(metrics, v1.ResourceCPU, watcher.CPU, node, nodeRequestsAndLimits)
+	riskMemory := pl.computeRisk(metrics, v1.ResourceMemory, watcher.Memory, node, nodeRequestsAndLimits)
+	rank := 1 - math.Max(riskCPU, riskMemory)
+
+	klog.V(6).InfoS("Node rank", "nodeName", node.GetName(), "riskCPU", riskCPU, "riskMemory", riskMemory, "rank", rank)
+
+	return rank
+}
+
+// computeRisk : calculate the risk of scheduling on node for a given resource
+func (pl *LowRiskOverCommitment) computeRisk(metrics []watcher.Metric, resourceName v1.ResourceName,
+	resourceType string, node *v1.Node, nodeRequestsAndLimits *trimaran.NodeRequestsAndLimits) float64 {
+	var riskLimit, riskLoad, totalRisk float64
+
+	defer func() {
+		klog.V(6).InfoS("Calculated risk", "node", klog.KObj(node), "resource", resourceName,
+			"riskLimit", riskLimit, "riskLoad", riskLoad, "totalRisk", totalRisk)
+	}()
+
+	nodeRequest := nodeRequestsAndLimits.NodeRequest
+	nodeLimit := nodeRequestsAndLimits.NodeLimit
+	nodeRequestMinusPod := nodeRequestsAndLimits.NodeRequestMinusPod
+	nodeLimitMinusPod := nodeRequestsAndLimits.NodeLimitMinusPod
+	nodeCapacity := nodeRequestsAndLimits.Nodecapacity
+
+	var request, limit, capacity, requestMinusPod, limitMinusPod int64
+	if resourceName == v1.ResourceCPU {
+		request = nodeRequest.MilliCPU
+		limit = nodeLimit.MilliCPU
+		requestMinusPod = nodeRequestMinusPod.MilliCPU
+		limitMinusPod = nodeLimitMinusPod.MilliCPU
+		capacity = nodeCapacity.MilliCPU
+	} else if resourceName == v1.ResourceMemory {
+		request = nodeRequest.Memory
+		limit = nodeLimit.Memory
+		requestMinusPod = nodeRequestMinusPod.Memory
+		limitMinusPod = nodeLimitMinusPod.Memory
+		capacity = nodeCapacity.Memory
+	} else {
+		// invalid resource
+		klog.V(6).InfoS("Unexpected resource", "resourceName", resourceName)
+		return 0
+	}
+
+	// (1) riskLimit : calculate overcommit potential load
+	if limit > capacity {
+		riskLimit = float64(limit-capacity) / float64(limit-request)
```

Comment:
Thanks @atantawi. Sorry for late response. The new simplified logic calculating risk sounds good. It's just could you make it UTed? in `TestLowRiskOverCommitment_computeRisk`.

Reply:
No problem, thanks